AURN_Data_Download/AURN_download.py

```diff
@@ -626,53 +665,65 @@ def test_preprocess_code(df_in,pt,spc_zero_process = ['O3','NO2','NOXasNO2'],min
-    
+    # get list of sites to processextract_site_data
+    if not site_list:
+        site_list = metadata['AURN_metadata']['site_id'].unique()
+    else:
+        # Todo - Test that sites provided are correct
+        pass
+    print('Site list', site_list)
+    #sys.exit(0)
+
     # create the station location dataset
     stations = metadata['AURN_metadata'][['site_id','latitude', 'longitude','site_name']].drop_duplicates()
     stations = stations.rename(columns={"site_id":"SiteID","latitude":"Latitude","longitude":"Longitude"})
     stations = stations.set_index('SiteID')
-    
-    
+
+
     # create a dataframe with the hourly dataset for all stations
-    #daily_dataframe = extract_site_data(site_list,metadata,years,data_path,save_to_csv)
-    hourly_dataframe = extract_site_data(site_list,metadata,years,data_path,save_to_csv)
-    hourly_dataframe= hourly_dataframe.rename(columns={'siteID':'SiteID'})
-    
+    #daily_dataframe = (site_list,metadata,years,data_path,save_to_csv)
+    hourly_dataframe = extract_site_data(site_list, metadata, years, data_path, save_to_csv)
+    hourly_dataframe = hourly_dataframe.rename(columns={'siteID':'SiteID'})
+
     # apply some filtering of negative and zero values
     spc_list = ['O3','PM10','PM2.5','NO2','NOXasNO2','SO2']
     for spc in spc_list:
-        print('{} has {} positive values'.format(spc,len(hourly_dataframe[spc][hourly_dataframe[spc]>0.0])))
-        print('{} has {} NaNs'.format(spc,len(hourly_dataframe[spc][hourly_dataframe[spc].isna()])))
-        print('{} has {} negative or zero values that will be replaced with NaNs'.format(spc,len(hourly_dataframe[spc][hourly_dataframe[spc]<=0.0])))
-        hourly_dataframe[spc][hourly_dataframe[spc]<=0.0]   = np.nan
+        try:
+            print('{} has {} positive values'.format(spc,len(hourly_dataframe.loc[hourly_dataframe[spc]>0.0])))
+            print('{} has {} NaNs'.format(spc,len(hourly_dataframe.loc[hourly_dataframe[spc].isna()])))
+            print('{} has {} negative or zero values that will be replaced with NaNs'.format(spc,len(hourly_dataframe.loc[hourly_dataframe[spc]<=0.0])))
+            hourly_dataframe.loc[hourly_dataframe[spc]<=0.0, 'spc'] = np.nan
+        except:
+            print('{} has  no values'.format(spc))
+
 
     # load the EMEP model data
-    if use_emep_data:
-        emep_dataframe = pd.read_csv(emep_file)
+    if emep_filename:
+        emep_dataframe = pd.read_csv(emep_filename)
         emep_dataframe = emep_dataframe.rename(columns={'NOx':'NOXasNO2'})
     else:
-        emep_dataframe = pd.DataFrame()
+        emep_dataframe = None
 
     # pull out the daily mean and max values for the site list
     # postprocessing the data set, to get daily data
-    daily_dataframe = postprocess_organisation(hourly_dataframe,emep_dataframe,stations,site_list,
-                                                   use_emep_data,impute_values)
+    useful_num_years = 0.8*len(years)
```

Comment:
I think we should have optional parameters for defining 'useful_num_years' and 'min_years' - and then move the code for calculating defaults for these based on the number of years chosen by the user into the routine itself.

Reply:
I've got this running, by setting the station_distances loop to go up to a maximum of the number of station distances, as it was going over that number before. I've also set a maximum value for station_number (but I don't think that was what broke it, but has potential to, if number of sites request by user is <5.

Reply:
But this needs your testing.

Reply:
Also, I'm not used to setting parameters for commandline apps, and followed your lead with the emep processing code.
I'm not sure I process the boolean arguments properly. I'll check/test these.

Reply:
These look fine to me. In addition I've added a check in the reading in of the parameters, so that useful_num_years is always greater or equal to min_years.